src/glum/_glm.py

```diff
@@ -1328,6 +1334,401 @@ def predict(
+        try:
+            test_stat = float(Rb_r.T @ linalg.solve(RVR, Rb_r))
+        except linalg.LinAlgError as err:
+            raise linalg.LinAlgError("The restriction matrix is not full rank") from err
+        p_value = 1 - stats.chi2.cdf(test_stat, Q)
+
+        return WaldTestResult(test_stat, p_value, Q)
+
+    def _wald_test_feature_name(
+        self,
+        features: Union[str, List[str]],
+        values: Optional[Sequence] = None,
+        X=None,
+        y=None,
+        mu=None,
+        offset=None,
+        sample_weight=None,
+        dispersion=None,
+        robust=None,
+        clusters: np.ndarray = None,
+        expected_information=None,
+    ) -> WaldTestResult:
+        """Compute the Wald test statistic and p-value for a linear hypothesis.
+
+        Perform a Wald test for the hypothesis that the coefficients of the
+        features in ``features`` are equal to the values in ``values``.
+
+        Parameters
+        ----------
+        features: Union[str, list[str]]
+            The name of a feature or a list of features to test.
+        values: Sequence, optional, default=None
+            The values to which coefficients are compared. If None, the test is
+            for whether the coefficients are zero.
+        X : {array-like, sparse matrix}, shape (n_samples, n_features), optional
+            Training data. Can be omitted if a covariance matrix has already
+            been computed.
+        y : array-like, shape (n_samples,), optional
+            Target values. Can be omitted if a covariance matrix has already
+            been computed.
+        mu : array-like, optional, default=None
+            Array with predictions. Estimated if absent.
+        offset : array-like, optional, default=None
+            Array with additive offsets.
+        sample_weight : array-like, shape (n_samples,), optional, default=None
+            Individual weights for each sample.
+        dispersion : float, optional, default=None
+            The dispersion parameter. Estimated if absent.
+        robust : boolean, optional, default=None
+            Whether to compute robust standard errors instead of normal ones.
+            If not specified, the model's ``robust`` attribute is used.
+        clusters : array-like, optional, default=None
+            Array with clusters membership. Clustered standard errors are
+            computed if clusters is not None.
+        expected_information : boolean, optional, default=None
+            Whether to use the expected or observed information matrix.
+            Only relevant when computing robust std-errors.
+            If not specified, the model's ``expected_information`` attribute is used.
+
+        Returns
+        -------
+        WaldTestResult
+            NamedTuple with test statistic, p-value, and degrees of freedom.
+        """
+
+        if isinstance(features, str):
```

Comment:
Do we need a check for whether features are None or an empty iterable?

Reply:
Hmm, good point. I'll test what happens with those inputs

Reply:
Calling the function with `None` returns a `TypeError: object of type 'NoneType' has no len()` exception, which is fine for non-end-users. End users using the public `wald_test` function should never encounter a case when this private function is called with `names=None`.

Calling it with an empty list returns a test-statistic of 0 (correct), and a p-value of `nan` (should be `1.0`, but I'm not sure it's worth special-casing it. I think these should make it obvious to the user what happened.

Reply:
Ok, thanks for checking.